poly/schnorr.go

```diff
@@ -0,0 +1,270 @@
+// When you want to verify a given signature :
+//  - You can just call from the schnorr struct : schnorr.VerifySignature(SchnorrSig, msg)
+// CAREFUL: your schnorr signature is a LONG TERM signature, you must keep the same throughout the differents rounds
+// otherwise you won't be able to verify any signatures. To be more precise, you just have to keep the same LongTerm sharedSecret
+// and PolyInfo. If you know these are the same throughout differents rounds, you can create many schnorr structs. This is
+// definitly NOT the way it is intented to be used, so use it at your own risks.
+type Schnorr struct {
+
+	// The info describing which kind of polynomials we using, on which groups etc
+	info PolyInfo
+
+	// The long-term shared secret evaluated by receivers
+	Longterm *SharedSecret
+
+	////////////////////////////////////////////////////
+	// FOR A GIVEN ROUND, we have the following members :
+
+	// hash is the hash of the message
+	hash *abstract.Secret
+
+	// The short term shared secret to use for this signature ONLY /!\
+	// i.e. the random secret in the regular schnorr signature
+	random *SharedSecret
+
+	// The partials signature of each other peers (i.e. receiver)
+	partials []*PartialSchnorrSig
+
+	/////////////////////////////////////////////////////
+}
+
+// Partial Schnorr Sig represents the partial signatures that each peers must generate in order to
+// generate the "global" signature. This struct must be sent across each peers for each peers
+type PartialSchnorrSig struct {
+	// The index of this partial signature regarding the global one
+	// same as the "receiver" index in the joint.go code
+	Index int
+
+	// The partial signature itself
+	Part *abstract.Secret
+}
+
+// SchnorrSig represents the final signature of a distribtued threshold schnorr signature
+// which can be verified against a message
+// This struct is not intended to be constructed manually but can be:
+//  - produce by the Schnorr struct
+//  - verified against a Schnorr struct
+type SchnorrSig struct {
+
+	// the signature itself
+	Signature *abstract.Secret
+
+	// the random public polynomial used during the signature generation
+	Random *PubPoly
+}
+
+// NewSchnorr nstantiate a Schnorr  struct . A wrapper around Init
+func NewSchnorr(info PolyInfo, longterm *SharedSecret) *Schnorr {
+	return new(Schnorr).Init(info, longterm)
+}
+
+// Init initialize the Schnorr struct
+func (s *Schnorr) Init(info PolyInfo, longterm *SharedSecret) *Schnorr {
+	s.info = info
+	s.Longterm = longterm
+	return s
+}
+
+// NewRound set the random key for the d.schnorr algo + set the msg to be signed
+// You call this function when you want a new signature to be issued on a specific message
+// The security of the d. schnorr signature protocol is the same as the regular :
+// The random secret MUST BE FRESH for EACH signature / signed message (hence the 'NewRound')
+func (s *Schnorr) NewRound(random *SharedSecret, msg []byte) error {
```

Comment:
Indeed. We have two different opinions on how we see this algorithm-flow. My approach looks more oriented towards a kind of distributed-schnorr-signature-as-a-service, where the matrix is already done, the longterm secret already computed, and each time we want to sign something, we just launch a new round. We expect this "service" to be used many times, so we don't have to recreate a Schnorr struct for each signing request. 
Yours looks more one-time fashion where we may or may not need to do the setup for each signing request, and we can sign something as long as we have both the longterm and the random shared secret. It is more useful when we know we don't have to accept many signing requests but rather on a one-time only manner.

Which approach is better than the other ? Linus, comments ? ;)